src/lib/comp/LayoutSearchFilter.svelte

```diff
@@ -0,0 +1,87 @@
+<script lang="ts">
+	import Input from '$lib/controls/Input.svelte';
+	import TextArea from '$lib/controls/TextArea.svelte';
+
+	export let id: string;
+	export let label: string = "Find:";
+	export let items: { [name: string]: any };
+	export let filterFunc: (itemKey:string, text:string) => boolean;
+	export let clearSearchFunc: () => void = () => {};
+	export let searchText: string = '';
+	export let title: string = '';
+	export let textArea: boolean = false;
+	export let autoExpand: boolean = false;
+	export let numResults: number = 0;
+	export let showNoneForBlank: boolean = false;
+	export let searching: boolean = false;
+	
+	let searchField: HTMLInputElement | null;
+	let rawSearchText: string = '';
+
+	function clearSearch() {
+		rawSearchText = '';
+		searchField = <HTMLInputElement>document.getElementById(id);
+		if (searchField) searchField.value = '';
```

Comment:
I think line 22 should clear the search field, so I don't think you need these lines.

Reply:
Actually no line 22 is not enough, that's why the extra lines, maybe because you have the value field in Input.svelte indirectly bound to the input's value. I didn't want to change your `display = (value: any) => value.toString()` because I didn't know what it does or what it would affect.